astropy/io/fits/tests/test_core.py

```diff
@@ -1068,7 +1155,7 @@ def get_free_space_in_dir(path):
             hdulist = fits.HDUList(hdu)
             filename = self.temp('test.fits')
 
-            with open(filename, mode='wb+') as fileobj:
+            with open(filename, mode='wb') as fileobj:
```

Comment:
Why this change ?

Reply:
This is a bit complicated: the short answer is that the previous use of `'wb+'` was incorrect, but the FITS file mode checking logic was not robust enough to detect the error.

The `writeto` method is used to create a new FITS file from scratch. This means the file is opened as an `'ostream'` in FITS terminology. To FITS, `'wb'` means `'ostream'`, whereas `'wb+'` means `'update'`. So the mode that the file was opened in was not in agreement with the way that FITS wanted to use it. Previously this error was allowed to pass unchecked, but with the more robust mode checking, it actually causes an error.

Reply:
Got it, thanks.

Reply:
Any chance this clarifies the question you asked above? I'm trying to write an explanation but keep failing at words 😛